src/tests/gtests.cpp

```diff
@@ -174,3 +198,61 @@ TEST_F(LeelaTest, MoveOnOccupiedSq) {
     // Find this error in the output
     EXPECT_NE(output.find("illegal move"), std::string::npos);
 }
+
+// Basic TimeControl test
+TEST_F(LeelaTest, TimeControl) {
+    auto maingame = get_gamestate();
+    std::string output;
+    // clear_board to force GTP to make a new UCTSearch.
+    // This will pickup our new cfg_* settings.
+    GTP::execute(maingame, "clear_board");
+
+    // Absolute time 100s = 1.37s per move in opening.
+    // Enough to be visible on display_times
+    GTP::execute(maingame, "kgs-time_settings canadian 0 120 25");
+    testing::internal::CaptureStderr();
+    GTP::execute(maingame, "showboard");
+    output = testing::internal::GetCapturedStderr();
+    expect_regex(output, "Black time: 00:02:00, 25 stones left");
+    expect_regex(output, "White time: 00:02:00, 25 stones left");
+
+    GTP::execute(maingame, "go");
+    testing::internal::CaptureStderr();
```

Comment:
This capture comes too late, which causes the search to get dumped to screen in-between the tests. (You can see this by running the test locally)

I would rather avoid this, i.e. capture all engine output during the tests, so you get the clean output of gtest only.

Reply:
fixed with new utility function.